charts/victoria-metrics-distributed/README.md

```diff
@@ -0,0 +1,283 @@
+```
+
+### How to roll out availability zone components?
+
+All the components under availability zone can be configured separately in `values.yaml`, it's recommend to upgrade one zone at a time, while stop ingesting and querying until the upgrade is completed.
+```
+availabilityZones:
+  # stop ingest and query from zone-a until upgrade is completed
+  - name: zone-a
+    # allow data ingestion to this zone
+    allowIngest: false
+    # allow data query from this zone through global query endpoint
+    allowQuery: false
+```
+
+## How to install
+
+Access a Kubernetes cluster.
+
+Add a chart helm repository with follow commands:
+
+```console
+helm repo add vm https://victoriametrics.github.io/helm-charts/
+
+helm repo update
+```
+
+List versions of ``vm/victoria-metrics-distributed``` chart available to installation:
+
+```console
+helm search repo vm/victoria-metrics-distributed` -l
+```
+
+Export default values of ``victoria-metrics-distributed``` chart to file ``values.yaml``:
+
+```console
+helm show values vm/victoria-metrics-distributed` > values.yaml
+```
+
+Change the values according to the need of the environment in ``values.yaml`` file.
+
+Test the installation with command:
+
+```console
+helm install vmcluster vm/victoria-metrics-distributed` -f values.yaml -n NAMESPACE --debug --dry-run
+```
+
+Install chart with command:
+
+```console
+helm install vmcluster vm/victoria-metrics-distributed` -f values.yaml -n NAMESPACE
+```
+
+Get the pods lists by running this commands:
+
+```console
+kubectl get pods -A | grep 'vminsert\|vmselect\|vmstorage'
+```
+
+Get the application by running this command:
+
+```console
+helm list -f vmcluster -n NAMESPACE
+```
+
+See the history of versions of ``vmcluster`` application with command.
+
+```console
+helm history vmcluster -n NAMESPACE
+```
+
+## How to uninstall
+
+Remove application with command.
+
+```console
+helm uninstall vmcluster -n NAMESPACE
+```
+
+## Documentation of Helm Chart
+
+Install ``helm-docs`` following the instructions on this [tutorial](../../REQUIREMENTS.md).
+
+Generate docs with ``helm-docs`` command.
+
+```bash
+cd charts/victoria-metrics-distributed`
+
+helm-docs
+```
+
+The markdown generation is entirely go template driven. The tool parses metadata from charts and generates a number of sub-templates that can be referenced in a template file (by default ``README.md.gotmpl``). If no template file is provided, the tool has a default internal template that will generate a reasonably formatted README.
+
+## Parameters
+
+The following tables lists the configurable parameters of the chart and their default values.
+
+Change the values according to the need of the environment in ``victoria-metrics-distributed`/values.yaml`` file.
+
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| availabilityZones[0].allowIngest | bool | `true` |  |
```

Comment:
Is it planned to add description to these params at some point?

Reply:
they were there, but with wrong comment format, fixed.